lib/gpu/gpuProcess.hpp

```diff
@@ -4,17 +4,26 @@
 #define HI_NIBBLE(b) (((b) >> 4) & 0x0F)
 #define LO_NIBBLE(b) ((b)&0x0F)
 
-#include "Stage.hpp"
-#include "gpuCommand.hpp"
-#include "gpuDeviceInterface.hpp"
-#include "gpuEventContainer.hpp"
-#include "restServer.hpp"
+#include "Stage.hpp"           // for Stage
+#include "bufferContainer.hpp" // for bufferContainer
 
-#include "json.hpp"
+#include <stdint.h> // for uint32_t
+#include <string>   // for string
+#include <thread>   // for thread
+#include <vector>   // for vector
+
+class gpuCommand;
```

Comment:
I'm not sure I understand what's going on with these forward declarations (here and in other places).   Why forward declare and not include?

Reply:
iwyu does that because it's supposed to compile faster (https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/WhyIWYU.md#why-forward-declare). There's an option to disable forward declares in a newer version of iwyu.

Reply:
I would be tempted to disable the forward declares option (adding `--no_fwd_decls`).  It seems like the current state will just cause problems when we try to use something referenced only as a forward declare.  And I think it will also cause issues with some IDEs autocomplete and documentation lookup.  Which is basically to say, I'm willing to trade a little compile time for not having to edit the includes when I use something that's only forward declared.